core/src/androidTest/java/com/bumble/appyx/core/plugin/BackPressHandlerTest.kt

```diff
@@ -0,0 +1,145 @@
+package com.bumble.appyx.core.plugin
+
+import android.app.Activity
+import androidx.activity.OnBackPressedCallback
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.test.espresso.Espresso
+import androidx.test.platform.app.InstrumentationRegistry
+import com.bumble.appyx.core.composable.Children
+import com.bumble.appyx.core.modality.BuildContext
+import com.bumble.appyx.core.node.ParentNode
+import com.bumble.appyx.core.node.node
+import com.bumble.appyx.debug.Appyx
+import com.bumble.appyx.routingsource.backstack.BackStack
+import com.bumble.appyx.routingsource.backstack.activeRouting
+import com.bumble.appyx.routingsource.backstack.operation.push
+import com.bumble.appyx.testing.ui.rules.AppyxTestRule
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.equalTo
+import org.hamcrest.Matchers.instanceOf
+import org.junit.After
+import org.junit.Rule
+import org.junit.Test
+
+class BackPressHandlerTest {
+
+    private var onBackPressedHandled = false
+    private var backPressHandler: BackPressHandler = object : BackPressHandler {
+        override val onBackPressedCallback: OnBackPressedCallback =
+            object : OnBackPressedCallback(true) {
+                override fun handleOnBackPressed() {
+                    onBackPressedHandled = true
+                }
+            }
+    }
+
+    @get:Rule
+    val rule = AppyxTestRule(launchActivity = false) { buildContext ->
+        TestParentNode(buildContext = buildContext, plugin = backPressHandler)
+    }
+
+    @After
+    fun after() {
+        Appyx.exceptionHandler = null
```

Comment:
Should this also be set in a before block? In case the exception handler was altered in a different test class?

Reply:
I think it is others test responsibility to clean the static state. Cleaning the static state everywhere seems hard to implement and maintain.

Reply:
fair enough